### PR TITLE
feat(panel): Allow slotting popover to attach to the header. #3196

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #3196

## Summary

feat(panel): Allow slotting popover to attach to the header. #3196

### CODE REVIEW

File: src/components/calcite-panel/calcite-panel.scss [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 :host {
   @extend %component-host;
-  @apply flex flex-auto overflow-hidden relative w-full;
```
Comment (driskull) [inline src/components/calcite-panel/calcite-panel.scss:13]: cant cut off popover

File: src/components/calcite-panel/calcite-panel.scss [hunk 2]
```diff
@@ -22,7 +22,8 @@
 @import "../../assets/styles/header";
 
 .container {
-  @apply flex flex-auto flex-col items-stretch bg-background w-full p-0 m-0;
+  @apply flex flex-auto flex-col items-stretch bg-background w-full p-0 m-0 relative;
+  z-index: 1;
```
Comment (driskull) [inline src/components/calcite-panel/calcite-panel.scss:26]: contains z-index of header.

**Comment by driskull** [comment]
@jcfranco can you review this one?

**Comment by jcfranco** [comment]
Apologies for the delayed review and thanks for taking a stab at this. This works for this specific use case, but I do wonder about the scalability and how it affects our other components.

I'd like to propose revisiting and to pivot towards improving the popover placement, so that it isn't cut off as described in the issue description.

Here are some reasons why I think we should reconsider:

* Popover placement is entirely dynamic, so even if we support the header as a reference element, there may still be inconsistent/undesirable placement.
* This would set precedent for adding slots for internal parts of components – this adds complexity to components that could be handled elsewhere.

Some ideas to try:

* Adding a way for users to provide modifiers/middleware to fine-tune positioning (we need to keep in mind the future move to `floating-ui` and how it could impact this).
* Introduce a custom modifier and component pattern for smarter placement (would need to sync with the design team to define this).

**Comment by driskull** [comment]
Sounds good. I'll close the PR

**Pull request closed without merging**